core/shared/src/main/scala/org/http4s/Entity.scala

```diff
@@ -22,26 +22,36 @@ import cats.~>
 import fs2.Chunk
 import fs2.Pure
 import fs2.Stream
+import org.http4s.headers.`Content-Length`
 
 sealed trait Entity[+F[_]] {
   def body: EntityBody[F]
-  def length: Option[Long]
+  def length: Option[`Content-Length`]
```

Comment:
Hmm. I'm not feeling so enjoyable with coupling `Entity` with a particular `Content-Length` header. Because we can easily infer that header from `Entity` using the current `length`. Probably having `ContentType` is more beneficial, as it does in `akka-http`. But yeah, that's out of the scope for this PR.

Reply:
I guess having `PosLong` is almost free. And it'd be pretty clear to end-users. The current `Entity` is fine-grained and that's good from an HTTP-domain modeling perspective. Also, we have a particular 'entity' that serves for aims of combining headers and `Entity` - `Media`.

Reply:
Typically any interesting model components have been in the root package, and headers have just been vessels for those.  I think this would be the first time a header was a "first class" value in the core package.

There are several places we need unsigned values.  Spire has a good one, but I don't want a Spire dependency just for `ULong`.  And many of them refer to lengths.  Perhaps introducing a `Length` type -- maybe with helpers for kilos and megas and what not -- would be good?